mkdocs_simple_plugin/plugin.py

```diff
@@ -5,11 +5,122 @@
 
 import fnmatch
 import os
+import re
 import shutil
 import sys
 import tempfile
 
 
+class LazyFile:
+    """
+    Like a file object, but only ever creates the directory and opens the
+    file if a non-empty string is written.
+    """
+    def __init__(self, direc, name):
+        self.fdir = direc
+        self.fname = name
+        self.fobj = None
+
+    def write(self, arg):
+        if arg == '':
+            return
+        if self.fobj is None:
+            os.makedirs(self.fdir, exist_ok=True)
+            self.fobj = open(f"{self.fdir}/{self.fname}", 'w')
+        self.fobj.write(arg)
+
+    def close(self):
+        if self.fobj is not None:
+            self.fobj.close()
+
+
+class StreamExtract:
+    """
+    Instantiating a StreamExtract object copies _instream_ to _outstream_,
+    extracting portions of the instream as specified by the keyword arguments
+    to the constructor as documented for the "semiliterate" parameter of the
+    `simple` plugin.
+    """
+    def __init__(self, instream, outstream,
+                 begin=None, end=None, pause=None, replace=[], incroot=None,
+                 **ignore_other_kwargs):
+        self.ins = instream
+        self.outs = outstream
+        self.begin = re.compile(begin)
+        self.end = (end is not None) and re.compile(end)
+        self.pause = (pause is not None) and re.compile(pause)
+        self.replace = []
+        for item in replace:
+            if isinstance(item, str):
+                self.replace.append(re.compile(item))
+            else:
+                self.replace.append((re.compile(item[0]), item[1]))
+        self.incroot = incroot
+        self.wroteSomething = False
```

Comment:
style: prefer underscores for instance variables

```suggestion
        self.wrote_something = False
```